manager/api/transport_test.go

```diff
@@ -0,0 +1,535 @@
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodGet, fmt.Sprintf("%s/clients", url), "", tc.auth, nil)
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			var data map[string][]manager.Client
+			json.NewDecoder(res.Body).Decode(&data)
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+			assert.ElementsMatch(t, tc.res, data["clients"], fmt.Sprintf("%s: expected body %s got %s", tc.desc, tc.res, data["clients"]))
+		}
+	})
+}
+
+func TestRemoveClient(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		svc.Register(validUser)
+		clientID, _ := svc.AddClient(validUser.Email, validClient)
+
+		cases := []struct {
+			desc     string
+			clientID string
+			auth     string
+			status   int
+		}{
+			{"delete existing client", clientID, validUser.Email, http.StatusNoContent},
+			{"try to delete non-existent client", "123e4567-e89b-12d3-a456-000000000042", validUser.Email, http.StatusNoContent},
+			{"try to delete client with invalid id", "1", validUser.Email, http.StatusNoContent},
+			{"try to delte client with invalid token", clientID, invalidUser.Email, http.StatusForbidden},
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodDelete, fmt.Sprintf("%s/clients/%s", url, tc.clientID), "", tc.auth, nil)
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		}
+	})
+}
+
+func TestCreateChannel(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		validData, _ := json.Marshal(validChannel)
+		svc.Register(validUser)
+
+		cases := []struct {
+			desc   string
+			req    string
+			auth   string
+			status int
+		}{
+			{"create new channel", string(validData), validUser.Email, http.StatusCreated},
+			{"try to create new channel with invalid token", string(validData), invalidUser.Email, http.StatusForbidden},
+			{"try to create new channel with invalid data format", "{", validUser.Email, http.StatusBadRequest},
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodPost, fmt.Sprintf("%s/channels", url), contentType, tc.auth, strings.NewReader(tc.req))
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		}
+	})
+}
+
+func TestUpdateChannel(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		update, _ := json.Marshal(map[string]string{
+			"name": "updated_channel",
+		})
+		svc.Register(validUser)
+		chanID, _ := svc.CreateChannel(validUser.Email, validChannel)
```

Comment:
Remarks given for client-related tests regarding asking to rename to `id` apply here as well.